In [67]:
%pip install pyodbc  
%pip install python-dotenv 

In [68]:
import pyodbc      #just installed with pip
from dotenv import dotenv_values    #import the dotenv_values function from the dotenv package
import pandas as pd
import warnings 

warnings.filterwarnings('ignore')

In [69]:
# Load environment variables from .env file into a dictionary
environment_variables = dotenv_values('.env')


# Get the values for the credentials you set in the '.env' file
server = environment_variables.get("SERVER")
database = environment_variables.get("DATABASE")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"

In [70]:
# Use the connect method of the pyodbc library and pass in the connection string.
# This will connect to the server and might take a few seconds to be complete. 
# Check your internet connection if it takes more time than necessary

connection = pyodbc.connect(connection_string)

In [71]:
# Reading the 2020 dataset. 

query = "Select * from LP1_startup_funding2020"

data2020 = pd.read_sql(query, connection)

In [72]:
# Reading the 2021 dataset. 

query = "Select * from LP1_startup_funding2021"

data2021 = pd.read_sql(query, connection)

In [73]:
# Reading the 2019 dataset. 

data2019 = pd.read_csv('startup_funding2019.csv')

In [74]:
data2018 = pd.read_csv('startup_funding2018.csv')

In [75]:
data2019.head(5)

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

datasets highlights, shape and composition


In [76]:
data2018.shape

(526, 6)

In [77]:
data2019.shape

(89, 9)

In [78]:
data2020.shape

(1055, 10)

In [79]:
data2021.shape

(1209, 9)

In [80]:
data2018.dtypes

Company Name     object
Industry         object
Round/Series     object
Amount           object
Location         object
About Company    object
dtype: object

the amount column need to be changed, 
we can investigate the company about to see if it has relevant information


In [81]:
data2018.head(5)

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

,,  ,the industry, round/series, amount, location columns need to be synchronized 
we should use columns in the 2018 data only

In [82]:
data2019.dtypes

Company/Brand     object
Founded          float64
HeadQuarter       object
Sector            object
What it does      object
Founders          object
Investor          object
Amount($)         object
Stage             object
dtype: object

 the founded, HeadQuarter, founders, investors column, do we need it?, what it does is like about company in 2018, stage is similar to round/series, 

In [83]:
data2019.head(5)

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [84]:
data2020.dtypes

Company_Brand     object
Founded          float64
HeadQuarter       object
Sector            object
What_it_does      object
Founders          object
Investor          object
Amount           float64
Stage             object
column10          object
dtype: object

In [85]:
data2020.head(10)

Company_Brand  Founded HeadQuarter               Sector  \
0    Aqgromalin   2019.0     Chennai             AgriTech   
1      Krayonnz   2019.0   Bangalore               EdTech   
2  PadCare Labs   2018.0        Pune   Hygiene management   
3         NCOME   2020.0   New Delhi               Escrow   
4    Gramophone   2016.0      Indore             AgriTech   
5        qZense   2019.0   Bangalore             AgriTech   
6  MyClassboard   2008.0   Hyderabad               EdTech   
7         Metvy   2018.0     Gurgaon  Networking platform   
8        Rupeek   2015.0   Bangalore              FinTech   
9     Gig India   2017.0        Pune        Crowdsourcing   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   
5  qZense Labs is building the next-generation Io...   
6  MyClassboard is a full-fledged School / Colleg...   
7  AI driven networking platform for individuals ...   
8  Rupeek is an online lending platform that spec...   
9  GigIndia is a marketplace that provides on-dem...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
5                       Rubal Chib, Dr Srishti Batra   
6                                     Ajay Sakhamuri   
7                                   Shawrya Mehrotra   
8            Amar Prabhu, Ashwin Soni, Sumit Maniyar   
9                       Aditya Shirole, Sahil Sharma   

                                        Investor      Amount         Stage  \
0                                Angel investors    200000.0          None   
1                                GSF Accelerator    100000.0      Pre-seed   
2                                 Venture Center         NaN      Pre-seed   
3            Venture Catalysts, PointOne Capital    400000.0          None   
4            Siana Capital Management, Info Edge    340000.0          None   
5  Venture Catalysts, 9Unicorns Accelerator Fund    600000.0          Seed   
6                                    ICICI Bank.    600000.0  Pre-series A   
7                                     HostelFund         NaN    Pre-series   
8   KB Investment, Bertelsmann India Investments  45000000.0      Series C   
9      Shantanu Deshpande, Subramaniam Ramadorai   1000000.0  Pre-series A   

  column10  
0     None  
1     None  
2     None  
3     None  
4     None  
5     None  
6     None  
7     None  
8     None  
9     None

In [86]:
data2021.dtypes

Company_Brand     object
Founded          float64
HeadQuarter       object
Sector            object
What_it_does      object
Founders          object
Investor          object
Amount            object
Stage             object
dtype: object

In [87]:
data2021.head(10)

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   
5   Urban Company   2014.0   New Delhi   Home services   
6  Comofi Medtech   2018.0   Bangalore      HealthTech   
7     Qube Health   2016.0      Mumbai      HealthTech   
8        Vitra.ai   2020.0   Bangalore    Tech Startup   
9          Taikee   2010.0      Mumbai      E-commerce   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   
5  Urban Company (Formerly UrbanClap) is a home a...   
6   Comofi Medtech is a healthcare robotics startup.   
7  India's Most Respected Workplace Healthcare Ma...   
8  Vitra.ai is an AI-based video translation plat...   
9  Taikee is the ISO-certified, B2B e-commerce pl...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   
5  Abhiraj Singh Bhal, Raghav Chandra, Varun Khaitan   
6                                         Gururaj KB   
7                                        Gagan Kapur   
8                                     Akash Nidhi PS   
9                 Nidhi Ramachandran, Sachin Chhabra   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed  
5                                     Vy Capital  $188,000,000          None  
6                              CIIE.CO, KIIT-TBI      $200,000          None  
7                      Inflection Point Ventures   Undisclosed  Pre-series A  
8                              Inflexor Ventures   Undisclosed          None  
9                                           None    $1,000,000          None

Assuming we will use company name, industry, type of funding, amount and location

Null hypothesis: The amount of funding a company receives is not affected by its location, type of industry, and the type of funding
alternative hypothesis; the amount of funding received by a company is significantly affected by the location of the company, type of funding and the industry the company is.

Questions:
1. The industry in which a startup operates is related to the amount of funding it receives
2. The location of a startup is related to the amount of funding it receives.
3. The funding type of the company is related to the amount the company receives in funding
4. The time when the startups were funded affects the amount the startups received.

Cleaning Data

In [88]:
data2019.isnull().sum()

Company/Brand     0
Founded          29
HeadQuarter      19
Sector            5
What it does      0
Founders          3
Investor          0
Amount($)         0
Stage            46
dtype: int64

In [89]:
data2018.isnull().sum()

Company Name     0
Industry         0
Round/Series     0
Amount           0
Location         0
About Company    0
dtype: int64

In [90]:
data2020.isnull().sum()

Company_Brand       0
Founded           213
HeadQuarter        94
Sector             13
What_it_does        0
Founders           12
Investor           38
Amount            254
Stage             464
column10         1053
dtype: int64

In [91]:
data2021.isnull().sum()

Company_Brand      0
Founded            1
HeadQuarter        1
Sector             0
What_it_does       0
Founders           4
Investor          62
Amount             3
Stage            428
dtype: int64

Given thata the 2018 data does not have null values, and the columns we are using for the 2018, 2019, 2020 and 2021 data have just a few missing variables, I will extract them then investigate them further.

In [92]:
data2018.describe()

Company Name Industry Round/Series Amount  \
count               526      526          526    526   
unique              525      405           21    198   
top     TheCollegeFever        —         Seed      —   
freq                  2       30          280    148   

                           Location  \
count                           526   
unique                           50   
top     Bangalore, Karnataka, India   
freq                            102   

                                            About Company  
count                                                 526  
unique                                                524  
top     TheCollegeFever is a hub for fun, fiesta and f...  
freq                                                    2

In [93]:
data2019.describe()

Founded
count    60.000000
mean   2014.533333
std       2.937003
min    2004.000000
25%    2013.000000
50%    2015.000000
75%    2016.250000
max    2019.000000

In [94]:
data2020.describe()

Founded        Amount
count   842.000000  8.010000e+02
mean   2015.363420  1.130430e+08
std       4.097909  2.476635e+09
min    1973.000000  1.270000e+04
25%    2014.000000  1.000000e+06
50%    2016.000000  3.000000e+06
75%    2018.000000  1.100000e+07
max    2020.000000  7.000000e+10

In [95]:
data2021.describe()

Founded
count  1208.000000
mean   2016.655629
std       4.517364
min    1963.000000
25%    2015.000000
50%    2018.000000
75%    2020.000000
max    2021.000000

a number of variables in the datasets have wrong data types and their summary statistics do not show up. They need to be changed

In [96]:
data2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


the amount variable need to be changed to an integer or float

In [97]:
data2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [98]:
data2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [99]:
data2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


renaming the columns before merging

In [100]:
#renaming 2018
data2018.rename(columns = {"Round/Series":"Stage"}, inplace = True)

In [101]:
#renaming 2019
data2019.rename(columns = {"Company/Brand":"Company Name","Sector":"Industry","What it does":"About Company","Amount($)":"Amount","HeadQuarter":"Location"}, inplace = True)
data2019.head(5)

Company Name  Founded   Location         Industry  \
0  Bombay Shaving      NaN        NaN        Ecommerce   
1       Ruangguru   2014.0     Mumbai           Edtech   
2        Eduisfun      NaN     Mumbai           Edtech   
3        HomeLane   2014.0    Chennai  Interior design   
4        Nu Genes   2004.0  Telangana         AgriTech   

                                       About Company  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor        Amount  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [102]:
#renaming 2020 data
data2020.rename(columns = {"Company_Brand":"Company Name","Sector":"Industry","What_it_does":"About Company","HeadQuarter":"Location"}, inplace = True)
data2020.sample(5)

Company Name  Founded   Location         Industry  \
87         Habitat   2016.0       None           EdTech   
438       BuildPan   2019.0     Indore     SaaS startup   
919    Paytm money   2010.0  Bangalore          Fintech   
271       LIVSPACE   2014.0  Bangalore  Interior Design   
181  Charcoal Eats   2015.0     Mumbai      QSR startup   

                                         About Company  \
87   Habitat, a social learning edtech platform for...   
438  Buildpan helps you with continuous build, deve...   
919                                  Wealth Management   
271  Livspace is an interior design startup that of...   
181  In the last five years, have served over 1.5 m...   

                                              Founders  \
87                         Rohit Pande, Shikhar Sachan   
438  Sonal Dandotia, Shantanu S, Virendra Chouhan, ...   
919                               Vijay Shekhar Sharma   
271  Anuj Srivastava, Ramakant Sharma, Shagufta Anurag   
181  Anurag Mehrotra, Krishnakant Thakur, Mohammed ...   

                                          Investor     Amount Stage column10  
87             Unitus Ventures, Whiteboard Capital   600000.0  Seed     None  
438              Sunil Kumar Singhvi, Yusho Kawata   500000.0  Seed     None  
919                           One97 Communications  5600000.0  None     None  
271  Mercer Investments, Bessemer Venture Partners  4000000.0  Debt     None  
181                             Lokmat Investments  2000000.0  None     None

In [103]:
#renaming 2021 columns
data2021.rename(columns = {"Company_Brand":"Company Name","Sector":"Industry","What_it_does":"About Company","HeadQuarter":"Location"}, inplace = True)
data2021.head(5)

Company Name  Founded   Location        Industry  \
0  Unbox Robotics   2019.0  Bangalore      AI startup   
1          upGrad   2015.0     Mumbai          EdTech   
2     Lead School   2012.0     Mumbai          EdTech   
3         Bizongo   2015.0     Mumbai  B2B E-commerce   
4        FypMoney   2021.0   Gurugram         FinTech   

                                       About Company  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

dropping irrelevant columns

In [104]:
nwdata2019 = data2019[['Company Name','Founded','Industry','About Company','Amount','Stage','Location']]
nwdata2019.tail(5)

Company Name  Founded                 Industry  \
84     Infra.Market      NaN                Infratech   
85              Oyo   2013.0              Hospitality   
86       GoMechanic   2016.0  Automobile & Technology   
87           Spinny   2015.0               Automobile   
88  Ess Kay Fincorp      NaN                  Banking   

                                        About Company        Amount     Stage  \
84  It connects client requirements to their suppl...   $20,000,000  Series A   
85                Provides rooms for comfortable stay  $693,000,000       NaN   
86  Find automobile repair and maintenance service...    $5,000,000  Series B   
87                                Online car retailer   $50,000,000       NaN   
88              Organised Non-Banking Finance Company   $33,000,000       NaN   

     Location  
84     Mumbai  
85   Gurugram  
86      Delhi  
87      Delhi  
88  Rajasthan

In [105]:
nwdata2020 = data2020[['Company Name','Founded','Industry','About Company','Amount','Stage','Location']]
nwdata2020.tail(5)

Company Name  Founded   Industry  \
1050  Leverage Edu      NaN     Edtech   
1051         EpiFi      NaN    Fintech   
1052       Purplle   2012.0  Cosmetics   
1053        Shuttl   2015.0  Transport   
1054         Pando   2017.0   Logitech   

                                          About Company      Amount  \
1050  AI enabled marketplace that provides career gu...   1500000.0   
1051  It offers customers with a single interface fo...  13200000.0   
1052         Online makeup and beauty products retailer   8000000.0   
1053                    App based bus aggregator serice   8043000.0   
1054            Networked logistics management software   9000000.0   

           Stage Location  
1050        None    Delhi  
1051  Seed Round     None  
1052        None   Mumbai  
1053    Series C    Delhi  
1054    Series A  Chennai

In [106]:
nwdata2021 = data2021[['Company Name','Founded','Industry','About Company','Amount','Stage','Location']]
nwdata2021.tail(5)

Company Name  Founded               Industry  \
1204      Gigforce   2019.0  Staffing & Recruiting   
1205        Vahdam   2015.0       Food & Beverages   
1206  Leap Finance   2019.0     Financial Services   
1207  CollegeDekho   2015.0                 EdTech   
1208        WeRize   2019.0     Financial Services   

                                          About Company     Amount  \
1204                  A gig/on-demand staffing company.   $3000000   
1205  VAHDAM is among the world’s first vertically i...  $20000000   
1206  International education loans for high potenti...  $55000000   
1207  Collegedekho.com is Student’s Partner, Friend ...  $26000000   
1208  India’s first socially distributed full stack ...   $8000000   

             Stage   Location  
1204  Pre-series A   Gurugram  
1205      Series D  New Delhi  
1206      Series C  Bangalore  
1207      Series B   Gurugram  
1208      Series A  Bangalore

In [107]:
#adding a year column to each dataset
data2018["Year"]= 2018

In [108]:
#adding a year column to each dataset
nwdata2019["Year"]= 2019

In [109]:
#adding a year column to each dataset
nwdata2020["Year"]= 2020

In [110]:
#adding a year column to each dataset
nwdata2021["Year"]= 2021

Merging the data

In [111]:
combineddata = pd.concat([data2018, nwdata2019, nwdata2020,nwdata2021], ignore_index=True)
combineddata.sample(10)

Company Name                                      Industry  \
2310      Convosight                                  SaaS startup   
2209           Trell                               Social commerce   
526   Bombay Shaving                                     Ecommerce   
2082          DeHaat                                      AgriTech   
2354      Green Cure                                    HealthCare   
1606       Tea Trunk                                      Beverage   
2240   PeakPerformer                                        EdTech   
379           Signzy  Computer, FinTech, Risk Management, Software   
2664          Fraazo                              Food & Beverages   
908       Fanball XI                                        Gaming   

             Stage        Amount                     Location  \
2310      Series A      $9000000                    New Delhi   
2209      Series B     $45000000                    Bangalore   
526            NaN    $6,300,000                          NaN   
2082      Series C   $30,000,000                        Patna   
2354  Pre-series A  $Undisclosed                    New Delhi   
1606          None      220000.0                          Goa   
2240    Early seed  $undisclosed                    Bangalore   
379       Series A  ₹240,000,000  Bangalore, Karnataka, India   
2664      Series B     $50000000                       Mumbai   
908           Seed     1000000.0                    New Delhi   

                                          About Company  Year  Founded  
2310  A mission - to help community builders become ...  2021   2019.0  
2209  India's Largest Lifestyle Social Commerce plat...  2021   2016.0  
526          Provides a range of male grooming products  2019      NaN  
2082  DeHaat connects farmers to suppliers and buyer...  2021   2012.0  
2354  World's first company to combine German Engine...  2021   2015.0  
1606                           Global premium tea brand  2020   2013.0  
2240  A goal-oriented, outcome-driven platform to ma...  2021   2020.0  
379   Signzy are creating 'building blocks for a Dig...  2018      NaN  
2664  FRAAZO is Mumbai's favourite App for Fresh Veg...  2021   2020.0  
908   The gaming company mostly focused on fantasy f...  2020      NaN

Cleaning the combined data

In [112]:
combineddata.nunique()

Company Name     2214
Industry          873
Stage              75
Amount            774
Location          172
About Company    2691
Year                4
Founded            34
dtype: int64

since we have 873 unique industries, I am going to re classify the industries to 5 major categories

In [124]:
# Standardize the "Stage" column to lowercase
combineddata['Stage'] = combineddata['Stage'].str.lower()

In [143]:
# Combine similar Seed categories
combineddata['Stage'].replace({'seed round': 'seed', 'seed funding': 'seed', 'seed+': 'seed', 'pre-seed round':'seed', 'pre seed round':'seed', 'Seed':'seed', 'pre-seed':'seed','early seed':'seed','seed a':'seed','seed round & series a':'seed','seed fund':'seed'}, inplace=True)

In [139]:
# Combine similar Series A categories
combineddata['Stage'].replace({'pre series a': 'series a', 'pre-series a': 'series a', 'pre-series a1': 'series a', 'seies a':'series a', 'series a+':'series a', 'post series a':'series a', 'pre- series a':'series a', 'pre series a1':'series a', 'series a-1':'series a','series a2':'series a'}, inplace=True)

In [141]:
# Combine similar Series B categories
combineddata['Stage'].replace({'pre-series b': 'series b', 'pre series b': 'series b', 'series b2': 'series b', 'series b3':'series b', 'series b+':'series b'}, inplace=True)

In [ ]:
# Combine similar all the other various types of funding to Others category
combineddata['Stage'].replace({'debt': 'other', 'angel': 'other', 'venture - series unknown': 'other', 'debt financing':'other', 'private equity':'other','corporate round':'other','bridge':'other','angel round':'other','grant':'other':'post-ipo equity':'other','secondary market':'other','pre-series':'other','$1200000':'other','undisclosed':'other','post-ipo debt':'other','$300000':'other','$6000000':'other','$1000000':'other',''}, inplace=True)

In [144]:
Stage_Vcounts = combineddata.Stage.value_counts().reset_index()
Stage_Vcounts.columns = ['stage','count']
Stage_Vcounts

stage  count
0                                                seed    768
1                                            series a    601
2                                            series b    156
3                                            series c    114
4                                            series d     50
5                                                debt     45
6                                               angel     37
7                            venture - series unknown     37
8                                            series e     30
9                                      debt financing     13
10                                           series f     11
11                                     private equity     10
12                                    corporate round      8
13                                             bridge      8
14                                           series h      4
15                                        angel round      4
16                                              grant      4
17                                    post-ipo equity      3
18                                   secondary market      3
19                                           series g      3
20                                         pre-series      3
21                                           $1200000      2
22                                        undisclosed      2
23                                      post-ipo debt      2
24                                       pre-series c      2
25                                            $300000      2
26                                          series d1      2
27                                           $6000000      1
28                                         mid series      1
29                                           $1000000      1
30                                          series f1      1
31                                                 pe      1
32                                          series f2      1
33                                        series c, d      1
34                                               edge      1
35                                    seed investment      1
36                                       pre series c      1
37                                       bridge round      1
38                                          series e2      1
39                                      fresh funding      1
40                                      funding round      1
41                              non-equity assistance      1
42  https://docs.google.com/spreadsheets/d/1x9zine...      1
43                                           series i      1